index.js

```diff
@@ -0,0 +1,201 @@
+const topscreen = document.getElementById("topscreen");
+const topscreen2 = document.getElementById("topscreen2");
+const quizCa = document.getElementById("category");
+const quizDi = document.getElementById("difficulty");
+const quizQu = document.getElementById("question");
+const answer = document.getElementById("answer");
+
+class Quiz {
+  constructor(quizData) {
+    this.quizData = quizData;
+  }
+  getCategory(num) {
+    return this.quizData[num].category;
+  }
+  getType(num) {
+    return this.quizData[num].type;
+  }
+  getDifficulty(num) {
+    return this.quizData[num].difficulty;
+  }
+  getQuestion(num) {
+    return this.quizData[num].question;
+  }
+  getCorrect_answer(num) {
+    return this.quizData[num].correct_answer;
+  }
+  getIncorrect_answers(num) {
+    return this.quizData[num].incorrect_answers;
+  }
+}
+
+const start = () => {
+  topscreen.innerHTML = "ようこそ";
+  // topscreen2.innerHTML = "以下のボタンをクリック";
+  quizQu.innerHTML = "以下のボタンをクリック";
+  const s_button = document.createElement("div");
+  answer.appendChild(s_button);
+  const s_button2 = document.createElement("button");
+  s_button.appendChild(s_button2);
+  s_button2.innerHTML = "スタート";
+  s_button2.style.backgroundColor = "lightgray";
+  s_button2.style.margin = "20px 0px";
+  s_button2.style.padding = "4px 4px";
+
+  return s_button2;
+};
+
+start().addEventListener("click", () => {
+  let num = 0;
+  let co_n = 0;
+
+  topscreen.innerHTML = "取得中";
+  quizQu.innerHTML = "少々お待ちください";
+  answer.innerHTML = "";
+  // console.log(topscreen);
+  // console.log(topscreen2);
+
+  fetch("https://opentdb.com/api.php?amount=10")
+    .then((res) => {
+      return res.json();
+    })
+    .then((res) => {
+      answer.innerHTML = "";
+      console.log(res);
+      // 10問分で初期化
+      const quiz = new Quiz(res.results);
+      const category = quiz.getCategory(num);
+      const type = quiz.getType(num);
+      const diffculty = quiz.getDifficulty(num);
+      const question = quiz.getQuestion(num);
+      const correct_answer = quiz.getCorrect_answer(num);
+      const incorrect_answers = quiz.getIncorrect_answers(num);
+
+      // クリックされるたびに、showQuizを実行させて、クイズを一問ずつ表示する
+      showQuiz(
+        category,
+        type,
+        diffculty,
+        question,
+        correct_answer,
+        incorrect_answers
+      );
+      answer.addEventListener("click", () => {
+        if (num < 10) {
+          showQuiz(
+            quiz.getCategory(num),
+            quiz.getType(num),
+            quiz.getDifficulty(num),
+            quiz.getQuestion(num),
+            quiz.getCorrect_answer(num),
+            quiz.getIncorrect_answers(num),
+            creatbutton(incorrect_answers, correct_answer)
+          );
+        } else {
+          creaetEndscreen();
+        }
+      });
+    });
+
+  const shuffleArray = (incorrect_answers, correct_answer) => {
+    const choices = incorrect_answers.concat(correct_answer);
```

Comment:
スプレッド演算子を使用すると、今風です！
```suggestion
    const choices = [...incorrect_answers, ...correct_answer];
```